proto/regen/ecocredit/marketplace/v1beta1/state.proto

```diff
@@ -106,24 +106,87 @@ message BuyOrder {
     // expiration is the optional timestamp when the buy order expires. When the
     // expiration time is reached, the buy order is removed from state.
     google.protobuf.Timestamp expiration = 8;
-}
 
-// AskDenom represents the information for an ask denom.
-message AskDenom {
 
+    // maker indicates that this is a maker order, meaning that when it hit
+    // the order book, there were no matching sell orders.
+    bool maker = 9;
+}
+
+// AllowedDenom represents the information for an allowed ask/bid denom.
+message AllowedDenom {
     option (cosmos.orm.v1alpha1.table) = {
-        id: 3,
-        primary_key: {fields: "denom"}
-        index: {id: 1 fields: "display_denom"}
+        id: 3
+        primary_key: {fields: "bank_denom"}
+        index: {id: 1 fields: "display_denom", unique: true}
     };
 
-    // denom is the denom to allow (ex. ibc/GLKHDSG423SGS)
-    string denom = 1;
+    // denom is the bank denom to allow (ex. ibc/GLKHDSG423SGS)
+    string bank_denom = 1;
 
-    // display_denom is the denom to display to the user and is informational
+    // display_denom is the denom to display to the user and is informational.
+    // Because the denom is likely an IBC denom, this should be chosen by
+    // governance to represent the consensus trusted name of the denom.
     string display_denom = 2;
 
     // exponent is the exponent that relates the denom to the display_denom and is
     // informational
     uint32 exponent = 3;
+}
+
+// Market describes a distinctly processed market between a credit type and
+// allowed bank denom. Each market has its own precision in the order book
+// and is processed independently of other markets. Governance must enable
+// markets one by one. Every additional enabled market potentially adds more
+// processing overhead to the blockchain and potentially weakens liquidity in
+// competing markets. For instance, enabling side by side USD/Carbon and
+// EUR/Carbon markets may have the end result that each market individually has
+// less liquidity and longer settlement times. Such decisions should be taken
+// with care.
+message Market {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 4
+        primary_key: {fields: "id"}
+        index: {id: 1 fields: "credit_type,bank_denom", unique: true}
+    };
+
+    // id is the unique ID of the market.
+    uint64 id = 1;
+
+    // credit_type is the abbreviation of the credit type.
+    string credit_type = 2;
+
+    // bank_denom is an allowed bank denom.
+    string bank_denom = 3;
```

Comment:
is this a base currency for all orders in this market? (eg "ust")?

Reply:
yes